server/odc-service/src/main/java/com/oceanbase/odc/service/datatransfer/task/DataTransferTask.java

```diff
@@ -0,0 +1,391 @@
+                .stream()
+                .filter(objectStatus -> objectStatus.getStatus() != Status.SUCCESS)
+                .map(ObjectResult::getSummary)
+                .collect(Collectors.toList());
+        Verify.verify(CollectionUtils.isEmpty(failedObjects),
+                "Data transfer task completed with unfinished objects! Details : " + failedObjects);
+    }
+
+    private void preHandleForImport(List<URL> inputs) throws IOException {
+        /*
+         * move import files
+         */
+        List<String> importFileNames = config.getImportFileName();
+        if (config.isCompressed()) {
+            DumperOutput dumperOutput = copyImportZip(importFileNames, workingDir);
+            /*
+             * set whiteList for ob-loader
+             */
+            List<DataTransferObject> objects = new ArrayList<>();
+            List<DumpDBObject> dumpDbObjects = dumperOutput.getDumpDbObjects();
+            for (DumpDBObject dbObject : dumpDbObjects) {
+                objects.addAll(dbObject.getOutputFiles().stream().map(abstractOutputFile -> {
+                    DataTransferObject transferObject = new DataTransferObject();
+                    transferObject.setDbObjectType(dbObject.getObjectType());
+                    transferObject.setObjectName(abstractOutputFile.getObjectName());
+                    return transferObject;
+                }).collect(Collectors.toList()));
+            }
+            config.setExportDbObjects(objects);
+
+        } else {
+            copyImportScripts(importFileNames, config.getDataTransferFormat(), workingDir, inputs);
+        }
+    }
+
+    private void postHandleForImport() throws FileNotFoundException {
+        if (workingDir == null || !workingDir.exists()) {
+            throw new FileNotFoundException("Working dir does not exist");
+        }
+        File importPath = Paths.get(workingDir.getPath(), "data").toFile();
+
+        if (importPath.exists()) {
+            boolean deleteRes = FileUtils.deleteQuietly(importPath);
+            log.info("Delete import directory, dir={}, result={}", importPath.getAbsolutePath(), deleteRes);
+        }
+        for (File subFile : workingDir.listFiles()) {
+            if (subFile.isDirectory()) {
+                continue;
+            }
+            boolean deleteRes = FileUtils.deleteQuietly(subFile);
+            log.info("Delete import file, fileName={}, result={}", subFile.getName(), deleteRes);
+        }
+    }
+
+    private void preHandleForExport() {
+        /*
+         * get mask config
+         */
+        String schemaName = config.getSchemaName();
+        Map<String, List<String>> tableName2ColumnNames = new HashMap<>();
+        ConnectionSession connectionSession = new DefaultConnectSessionFactory(connectionConfig).generateSession();
+        try {
+            DBSchemaAccessor accessor = DBSchemaAccessors.create(connectionSession);
```

Comment:
use schema extension~

Reply:
we need `accessor.listTableColumns` here, the schema extension don't have this method yet... If using `listTable`, there will be wasted queries.